### DISCUSSION TITLE
Interferograms processed by stripmapApp are smaller than the same processed by insarApp

### DISCUSSION THREAD

**Discussion by fdelgadodelapuente** [isce-framework/isce2#586] [General]
Dear ISCE developers,

I've noticed that  ENVISAT, ALOS and  COSMO-SkyMed interferograms processed from raw data by stripmapApp.py have a smaller swath (40x40 km2) than the same datasets processed by the deprecated insarApp.py (50x50 km2). Is it due to the different focusing algorithms in the workflows or to some other difference?

Thanks
Francisco

**Comment by EJFielding** [comment]
Hi Francisco,
Yes, the two processing systems have different focusing algorithms and different parameters. The old `insarApp.py` used a motion-compensation algorithm to focus the two scenes of raw data as though they were both acquired on a reference track. I think the parameters for the focusing in `insarApp` extended the swath to attempt to get all the data even if the baseline between the actual and reference track is long. We used to see a lot of ugly bad phase along the edges with the COSMO-SkyMed interferograms processed with `insarApp` because it was extending the radar pulse compression beyond where there is useful data. I expect there was similar extension in the azimuth direction. You can probably see that the radar amplitude decreases in the near and far range due to the range extension with a partial pulse.
The `stripmapApp.py` program uses a basic range-Doppler focusing algorithm with the original orbit track and Doppler centroid, similar to the "roi" program in the old ROI_pac package. It probably has the range and azimuth extension parameters set to smaller values because it does not need to process more area to get all the data.

**Comment by fdelgadodelapuente** [comment]
Thanks Eric for the detailed explanation. 

Here is an example of a COSMO-SkyMed Interferogram of Villarrica volcano in southern Chile and you can see that the insarApp swath (dark amplitude and phase) is ~10 km longer in the ground range direction than the stripmapApp swath (on top).  There is almost no difference in the azimuth direction. 

![csk](https://user-images.githubusercontent.com/23100180/190882322-56e8784c-8603-44d1-be9b-61184efa7d44.png)

Do you know how to change the range extension parameters in stripmapApp?

Thanks
Francisco

**Reply by EJFielding** [reply]
I don't know how to change the range extension parameters in `stripmapApp`. I think those parameters are not exposed for modification in the `stripmapApp.xml`. I remember spending some time trying to reduce the range extension in `insarApp` back around 2014 because the phase at the extreme edges (where the amplitude is very dark) was often some kind of noise that ends up being coherent in interferograms and messes up the phase unwrapping. The way that `insarApp` and the modules it calls were written, it was not easy to change the extension without a lot of rewriting.

**Reply by fdelgadodelapuente** [reply]
Thanks Eric. Actually the range extension parameters are not part of the stripmapApp inputs.  I'll have to dive into the software internals and try to change them.

Francisco

**Reply by EJFielding** [reply]
The ISCE2 code is structured in a way that makes changing things like that quite difficult. Not like the old ROI_pac code.

**Reply by fdelgadodelapuente** [reply]
Do you know a paper where  the range extension concept is described in detail?

**Reply by EJFielding** [reply]
I don't know the SAR signal processing literature well. It is using partial chirp compression where the program uses a portion of the reference chirp to match the signal data past the point where the full chirp is present. A paper on range compression methods should explain this.

**Comment by kangahdesmond** [comment]
Please can i get the steps in processing sentinel 1 image using stripmapApp.py in a command line please 🙏,  I am a master student at the Louisiana state university and i am new to this research area and thats my research interests,  my email is dkanga1@lsu.edu,  i would appreciate your feedback, many thanks

**Reply by EJFielding** [reply]
Hello Kangahdesmond,
Is your Sentinel-1 data from the Interferometric Wideswath (IW) mode, also called TOPS, or the stripmap mode? Most Sentinel-1 is IW or TOPS, and must be processed with `topsApp.py`.

**Comment by kangahdesmond** [comment]
My data is Sentinel One, and its IW, and I have called the topsApp.py --help, and it works, as well as the stripmapApp.py --help, and it also works. I am a master's student, and I would be working on 100 Sentinel One images, and I am new to this area and would need your guidance to start.
________________________________

**Reply by EJFielding** [reply]
Hello, I would recommend looking at the training materials including videos from the EarthScope InSAR with ISCE+ Short Course in 2024 and earlier years.

**Comment by kangahdesmond** [comment]
Sure I have started watching the videos and reading the materials and i would like to join the EarthScope InSAR 2025

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
